dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/DefaultTrackerProgramRuleService.java

```diff
@@ -79,7 +82,12 @@ public DefaultTrackerProgramRuleService(
             .collect( Collectors.toMap( Enrollment::getEnrollment, e -> {
                 ProgramInstance enrollment = enrollmentTrackerConverterService.fromForRuleEngine( bundle.getPreheat(),
                     e );
-                return programRuleEngine.evaluate( enrollment, Sets.newHashSet() );
+                try{
+                    return programRuleEngine.evaluate( enrollment, Sets.newHashSet() );
+                } catch (Exception ex) {
+                    log.warn("Something bad happen calling rule engine. The reason should be an invalid payload");
```

Comment:
Just a suggestion. Instead of hiding the original exception, wouldn't it be better to use `log.warn( "message", ex);`

Reply:
Of course. Just got lost in the commit.
Thanks